frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -342,6 +342,18 @@ function ReaderRolling:onCloseDocument()
             logger.dbg("cre DOM may not be in sync with styles, invalidating cache file for a full reload at next opening")
             self.ui.document:invalidateCacheFile()
         end
+        -- 'title' and 'authors' props are saved to cre cache on first document opening.
+        -- If the document has no 'title', crengine saves filename-without-extension as 'title'.
+        -- If the props have been changed before first closing the document,
+        -- restore original values to write them to the cache.
```

Comment:
Oh, I didn't thought about this having any effect on the cache (otherwise, I would have shut up :)).
I'm not sure I get what this section is about though.
If we override title & authors, does that make the cache invalid, and so rewritten?
Or is that restoring actually to avoid saving/rewritting a cache ?
Also, sure title or splitFileNameType() always return exactly what crengine uses internally, and that there is no trimming that could make them different?

Also, remind me:
`self.ui.doc_settings:readSetting("title")` is always the original title ?
and doc_props.title is the overriden or original one?

Reply:
`doc_settings:readSetting("title")` is the original title from the document, can be `nil`.
`doc_props.title` is the customized title.
I tested with a txt file, with various names, with trimming spaces, works good.

The cache file is written only once, at first document opening (or until invalidation).
Changing the document props via new `overrideDocumentProp()` does not change the cache, the props in cache are not overridden. So the mentioned piece of code is needed on first opening only (and after invalidation), but I cannot detect the **first** opening, so it will be executed on every opening/closing.
Techically: the cache file is created on opening the document, and flushed on closing the document.

Reply:
OK, I get it I think.
I quickly checked, and indeed, document props does not influence the rendering hash, so no rerendering if we change title/authors. Good.
They are indeed saved in, and loaded from, the cache.
And you want to always keep the original genuine ones in the cache. I agree.

I think there may be an issue with partial rerenderings, where the rerendering and new cache is saved in the subprocess, so probably with crengine having the customized metadata. Is this the case ?

It feels a bit sad to have to do this gymnastic with restoring original metadata. I think I'd rather prefer to have to hack LVDocView to store internally alt_authors and alt_title, and use that when drawing the top status bar if they are set, without ever touching its own DocumentProps...

Reply:
And I have checked, actually crengine trims starting/ending spaces from the filename to use it when there is no title.
I think we can follow it, for producing our synthetic title.